pkg/worker/handler/operator/ensure.go

```diff
@@ -41,12 +37,30 @@ func (h *Handler) Ensure() error {
 		}
 	}
 
-	// TODO
+	// Once the current and desired states of the runnable service releases are
+	// known, we can run the next steps in parallel too. Additionally, we only
+	// need to do real work in those following steps, if we recognize any state
+	// drift.
 	//
-	//     check for drift
-	//     apply update, if any
-	//     emit deployment event, if updated
+	//     1. Check whether those ECR image tags exist that are specified in the
+	//        desired state of any given service release.
 	//
+	//     2. TODO fetch existing cloudformation templates from infrastructure repo
+	//
+
+	{
+		err = parallel.Func(h.reg.Ensure /* , h.inf.Ensure */)
+		if err != nil {
+			return tracer.Mask(err)
+		}
```

Comment:
The step missing here now is to lookup the cloudformation templates from our infrastructure repository, which we only have to do if there is at least one state drift.